File: app/src/main/java/com/example/sarahn/waterreminderapp/classes/ClsTimePickerDialogBuilder.java
package com.example.sarahn.waterreminderapp.classes;

import android.content.Context;
import android.util.Log;
import android.widget.Toast;

import com.borax12.materialdaterangepicker.time.RadialPickerLayout;
import com.borax12.materialdaterangepicker.time.TimePickerDialog;

import java.util.Calendar;

/**
 * Created by SarahN on 6/18/2017.
 */
public class ClsTimePickerDialogBuilder{

    public static int fromHour;
    public static int fromMin;
    public static int toHour;
    public static int toMin;
    public static int timeDifference;


    public static TimePickerDialog showDialog(final Context context){
        Calendar now = Calendar.getInstance();

        TimePickerDialog tpd;
        return tpd = TimePickerDialog.newInstance(new TimePickerDialog.OnTimeSetListener() {
                                                      @Override
                                                      public void onTimeSet(RadialPickerLayout radialPickerLayout, int i, int i1, int i2, int i3) {
                                                          fromHour = i;
                                                          fromMin = i1;
                                                          toHour = i2;
                                                          toMin = i3;

                                                        ClsComfirmationDialog.displayDialog(context).show();
                                                          timeDifference = (fromMin - toMin) + ( fromHour - toHour);
                                                          timeDifferenceUtil(timeDifference, context);
                                                      }}, now.get(Calendar.HOUR_OF_DAY),
                now.get(Calendar.MINUTE),
                false);
    }


    private static void timeDifferenceUtil(float td, Context context){
        if(timeDifference > 0){
            ClsComfirmationDialog.displayWarningDialog(context).show();
        }else if(Math.abs(timeDifference) <= 2 ){
            ClsComfirmationDialog.displayWarningDuration(context).show();
        }
    }

}

File: app/src/main/java/com/example/sarahn/waterreminderapp/dialogs/ClsTimePickerDialogBuilder.kt
package com.example.sarahn.waterreminderapp.dialogs

import android.content.Context
import com.borax12.materialdaterangepicker.time.TimePickerDialog
import com.example.sarahn.waterreminderapp.Utils.Logging
import com.example.sarahn.waterreminderapp.Utils.NotificationUtills
import com.example.sarahn.waterreminderapp.Utils.SharedPrefUtils
import com.example.sarahn.waterreminderapp.classes.MyApplication
import com.example.sarahn.waterreminderapp.dialogs.ClsComfirmationDialog.displayDialog
import com.example.sarahn.waterreminderapp.dialogs.ClsComfirmationDialog.displayWarningDuration
import java.util.*

/**
 * Created by SarahN on 6/18/2017.
 */
object ClsTimePickerDialogBuilder {
    var fromHour = 0
    var fromMin = 0
    var toHour = 0
    var toMin = 0
    @JvmField
    var timeDifference = 0
    @JvmStatic
    fun showDialog(context: Context?): TimePickerDialog {
        val now = Calendar.getInstance()
        val tpd: TimePickerDialog
        tpd = TimePickerDialog.newInstance({ radialPickerLayout, i, i1, i2, i3 ->
            fromHour = i
            fromMin = i1
            toHour = i2
            toMin = i3
            Logging.logMessage("times " + fromHour + toHour + fromMin + toMin)
            SharedPrefUtils.setStartHour(MyApplication.context, i)
            SharedPrefUtils.setEndHour(MyApplication.context, i2)
            SharedPrefUtils.setStartMin(MyApplication.context, i1)
            SharedPrefUtils.setEndMin(MyApplication.context, i3)
            displayDialog(context!!).show()
            val hours = (toHour - fromHour) * 60
            val min = toMin - fromMin
            timeDifference = hours + min
            Logging.logMessage("time difference " + timeDifference)
            NotificationUtills.NotificationCounter()

            //  timeDifference = (toHour - fromHour) - (toMin - fromMin);
            // timeDifferenceUtil(timeDifference, context);
        }, now[Calendar.HOUR_OF_DAY],
                now[Calendar.MINUTE],
                true)
        return tpd
    }

    private fun timeDifferenceUtil(td: Float, context: Context) {
        if (timeDifference > 0) {
            //   ClsComfirmationDialog.displayWarningDialog(context).show();
        } else if (Math.abs(timeDifference) <= 2) {
            displayWarningDuration(context).show()
        }
    }
}